modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxStateMeta.java

```diff
@@ -51,13 +55,54 @@ public TxStateMeta(
             String txCoordinatorId,
             @Nullable TablePartitionId commitPartitionId,
             @Nullable HybridTimestamp commitTimestamp
+    ) {
+        this(txState, txCoordinatorId, commitPartitionId, commitTimestamp, 0);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param txState Transaction state.
+     * @param txCoordinatorId Transaction coordinator id.
+     * @param commitPartitionId Commit partition replication group id.
+     * @param commitTimestamp Commit timestamp.
+     * @param lastAbandonedMarkerTs Timestamp indicates when the transaction is marked as abandoned.
+     */
+    private TxStateMeta(
+            TxState txState,
+            String txCoordinatorId,
+            TablePartitionId commitPartitionId,
+            HybridTimestamp commitTimestamp,
+            long lastAbandonedMarkerTs
     ) {
         this.txState = txState;
         this.txCoordinatorId = txCoordinatorId;
         this.commitPartitionId = commitPartitionId;
         this.commitTimestamp = commitTimestamp;
     }
 
+    /**
+     * Creates a transaction state for the same transaction, but this one is marked abandoned.
+     *
+     * @return Transaction state meta.
+     */
+    public TxStateMetaAbandoned markAbandoned() {
+        assert checkTransitionCorrectness(txState, ABANDONED) : "Transaction state is incorrect [txState=" + txState + "].";
+
+        return new TxStateMetaAbandoned(txCoordinatorId, commitPartitionId);
+    }
+
+    /**
+     * Creates a transaction state for the same transaction, but this one is marked finishing.
+     *
+     * @return Transaction state meta.
+     */
+    public TxStateMetaFinishing markFinishing() {
+        assert checkTransitionCorrectness(txState, FINISHING) : "Transaction state is incorrect [txState=" + txState + "].";
+
+        return new TxStateMetaFinishing(txCoordinatorId, commitPartitionId);
+    }
+
```

Comment:
I mean, we use the same naming in *Listener classes, to actually mark the transactions with some state in the local state map. Here it is not marking, it's a creation of a new meta. Let's name it simply 'abandoned` and `finishing`.

Reply:
Renamed.